src/components/Authorization.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { FC, ReactNode } from 'react';
+import { useFusionAuthContext } from '../providers/FusionAuthProvider';
+
+interface Props {
+    authorizedRole?: string;
+    children?: ReactNode;
+}
+
+export const Authorization: FC<Props> = ({ authorizedRole, children }) => {
+    const { user } = useFusionAuthContext();
+
+    const isAuthorized = authorizedRole
+        ? Object.keys(user).length !== 0 && user.role === authorizedRole
+        : Object.keys(user).length !== 0;
+
+    return <>{isAuthorized && children}</>;
+};
```

Comment:
I think we should rename this component to something like `RequireAuth`, and then change the role prop to something like `withRole`. Then usage looks like 
```
<RequireAuth withRole="admin"> ... </RequireAuth>
```